django_loci/tests/__init__.py

```diff
@@ -94,6 +94,20 @@ def _login_as_admin(self):
         self.client.force_login(admin)
         return admin
 
+    def _create_readonly_admin(self, **kwargs):
+        """Creates a read-only admin user with view permissions for the specified models."""
+        #
+        from django.contrib.auth.models import Permission
+
+        permission_codename = []
+        for model in kwargs.pop('models', []):
+            permission_codename.append(f'view_{model.__name__.lower()}')
+        # assign view permissions to user
+        view_permission = Permission.objects.filter(codename__in=permission_codename)
+        user = self._create_admin(is_superuser=False, **kwargs)
+        user.user_permissions.add(*view_permission)
```

Comment:
The user should be created first, and then permissions can be assigned. This is because the kwargs models list can be empty, and adding view permissions is not always necessary when the queryset is empty.

Reply:
One more question - do we really need to pass models to this function? Can’t we automatically assign all read-only model permissions to the user instead?

Reply:
> One more question - do we really need to pass models to this function? Can’t we automatically assign all read-only model permissions to the user instead?

I was thinking of providing a fine grained control over permissions so as to simulate the working for a user with view-only access to specific models. Let me know if this requires any changes